include/volume/volume_cooling_gaussians.hpp

```diff
@@ -335,11 +335,22 @@ double volume_cooling_gaussians(Polytope& Pin,
     NT C = parameters.C;
     unsigned int N = parameters.N;
 
-    compute_annealing_schedule
-    <
-        WalkType,
-        RandomPointGenerator
-    >(P, ratio, C, parameters.frac, N, walk_length, radius, error, a_vals, rng);
+    // Construct the necesary parameters if this is the case
+    // Construct the necessary parameters if this is the case
+    if constexpr (std::is_same<WalkTypePolicy, GaussianHamiltonianMonteCarloExactWalk>::value) {
+        if (L > 0) {
+            typename GaussianHamiltonianMonteCarloExactWalk::parameters walk_params(L, true, 0, false);
+            compute_annealing_schedule<GaussianHamiltonianMonteCarloExactWalk, RandomPointGenerator>(
+                P, ratio, C, parameters.frac, N, walk_length, radius, error, a_vals, rng, walk_params);
+        } else {
+            compute_annealing_schedule<WalkType, RandomPointGenerator>(
+                P, ratio, C, parameters.frac, N, walk_length, radius, error, a_vals, rng);
+        }
+        } else {
```

Comment:
duplicate code?

Reply:
actually, not really. It is just bad indexation. As there is a run time and compile time check (with the L, and the gaussian type policy). If there is a gaussian type walk, I want the walk to check if L>0 or not.